python/setup.py

```diff
@@ -199,7 +199,7 @@ def _supports_symlinks():
         extras_require={
             'ml': ['numpy>=1.7'],
             'mllib': ['numpy>=1.7'],
-            'sql': ['pandas']
+            'sql': ['pandas>=0.13.0']
```

Comment:
Thanks for researching this @hyukjinkwon!  I opened a follow-up to add more type support.  I can do related docs there and we could also discuss whether or not to add pyarrow to the setup.py file once that's complete.